estate/models/estate_property.py

```diff
@@ -0,0 +1,97 @@
+from datetime import date
+from dateutil.relativedelta import relativedelta
+from odoo.exceptions import UserError
+from odoo import api, fields, models
+
+
+class EstateProperty(models.Model):
+    _name = "estate.property"
+    _description = "Estate property"
+    _order = ""
+
+    name = fields.Text('Title', required=True)
+    description = fields.Text('Description')
+    post_code = fields.Char('Postcode')
+    date_availability = fields.Date(
+        'Available From',
+        copy=False,
+        default=lambda self: date.today() + relativedelta(months=3),
+    )
+    expected_price = fields.Float('Expected Price', required=True)
+    selling_price = fields.Float('Selling Price', readonly=True, copy=False)
+    bedrooms = fields.Integer('Bedrooms', default=2)
+    living_area = fields.Integer('Living Area (sqm)')
+    facades = fields.Integer('Facades')
+    garage = fields.Boolean('Garage')
+    garden = fields.Boolean('Garden')
+    garden_area = fields.Integer('Garden Area (sqm)')
+    garden_orientation = fields.Selection(
+        selection=[
+            ('north', 'North'),
+            ('south', 'South'),
+            ('east', 'East'),
+            ('west', 'West'),
+        ],
+        string='Garden Orientation',
+    )
+    state = fields.Selection(
+        selection=[
+            ('new', 'New'),
+            ('offer', 'Offer'),
+            ('received', 'Received'),
+            ('offer_accepted', 'Offer Accepted'),
+            ('sold', 'Sold'),
+            ('cancelled', 'Cancelled'),
+        ],
+        string='State',
+        default='new',
+        required=True,
+        copy=False,
+    )
+    active = fields.Boolean('Active', default=True)
+    property_type_id = fields.Many2one("estate.property.type", string="Property Type")
+    user_id = fields.Many2one(
+        'res.users', string='Salesman', default=lambda self: self.env.user
+    )
+    buyer_id = fields.Many2one("res.partner", string="Buyer", readonly=True, copy=False)
+    tag_ids = fields.Many2many("estate.property.tag", string="Tags")
+    offer_ids = fields.One2many("estate.property.offer", "property_id", string="Offers")
+    total_area = fields.Integer("Total Area (sqm)", compute="_compute_total_area")
+    best_price = fields.Float(
+        "Best Offer", compute="_compute_best_price", readonly=True
+    )
+
+    @api.depends('living_area', 'garden_area')
+    def _compute_total_area(self):
+        for rec in self:
+            rec.total_area = (rec.living_area or 0) + (rec.garden_area or 0)
+
+    @api.depends('offer_ids.price')
+    def _compute_best_price(self):
+        for rec in self:
+            prices = rec.offer_ids.mapped('price')
+            rec.best_price = max(prices) if prices else 0.0
```

Comment:
That works, you can also do 
```suggestion
            rec.best_price = max(prices, default=0.0)
```